### PR TITLE
ActionList: Enable focusZone for roles listbox and menu

### PR DESCRIPTION
**Pull request by siddharthkp**
- Fixes https://github.com/github/primer/issues/3475

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [x] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

**Comment by siddharthkp** [comment]
Identified some of the failing tests in dotcom as coming from the release candidate. Will wait for next dotcom upgrade before going too deep in debugging

File: packages/react/src/ActionList/List.tsx
Comment (TylerJDev) [inline packages/react/src/ActionList/List.tsx:50]: I'm wondering if we'd want to add `focusOutBehavior: 'wrap'` by default, or allow consumers to add that themselves?

File: packages/react/src/ActionList/List.tsx
Comment (TylerJDev) [inline packages/react/src/ActionList/List.tsx:48]: Non-blocking: I see we have have `menubar`, I am wondering if there'd be a case where `role=menubar` is used with ActionList 🤔

File: packages/react/src/ActionList/List.tsx
Comment (TylerJDev) [inline packages/react/src/ActionList/List.tsx:46]: If there's an existing focus zone being used for an `ActionList`, would the one being used here replace it? I saw an instance where there was an `ActionList` inside of an `AnchoredOverlay` that had an implicit focus zone attached: 

```tsx
<AnchoredOverlay
  renderAnchor={props => <Button {...props}>Button</Button>}
  focusZoneSettings={{
    focusOutBehavior: 'wrap',
  }}
>
  <ActionList role="listbox">
    <ActionList.Item>Item 1</ActionList.Item>
    <ActionList.Item>Item 2</ActionList.Item>
  </ActionList>
</AnchoredOverlay>
```
I'm wondering if this would be an issue or not 🤔. I think the existing cases are rare, so doesn't seem like it would be.

**Review by TylerJDev — APPROVED** [review]
LGTM! Few non-blocking comments!

File: packages/react/src/ActionList/List.tsx
Comment (siddharthkp) [reply packages/react/src/ActionList/List.tsx:48]: We don't have one yet, but I think it's possible. The [issue linked to menu + menubar pattern](https://arc.net/l/quote/brrnmqyb), so I included it just in case

File: packages/react/src/ActionList/List.tsx
Comment (siddharthkp) [reply packages/react/src/ActionList/List.tsx:46]: focus zone can only be added, not removed. So, if the parent AnchoredOverlay has a focus zone, that would work on all it's children. Setting it to false here would not disable to parent. (ActionMenu, Autocomplete, etc. also use AnchoredOverlay)

File: packages/react/src/ActionList/List.tsx
Comment (siddharthkp) [reply packages/react/src/ActionList/List.tsx:50]: > I'm wondering if we'd want to add focusOutBehavior: 'wrap' by default

Honestly, not sure, what do you think?

> or allow consumers to add that themselves?

That's the part that's missing right now. I think I'd like to wait for feedback/requests before introducing `focusZoneSettings` like we have on other focus-zoning components

Comment (TylerJDev) [reply packages/react/src/ActionList/List.tsx:50]: For `role=menu`, we'd definitely want it to wrap by default. With `role=listbox` there may be cases where it doesn't need to wrap. I'd say if we can conditionally set it to `wrap` based on if it's a menu or not would be a safe bet, at least for now!

Comment (siddharthkp) [reply packages/react/src/ActionList/List.tsx:50]: Done!

**Comment by siddharthkp** [comment]
Updated focusOutBehavior to wrap, will test with dotcom once again before merging

**Pull request merged**
